widget/embedded/src/components/CustomDestination/CustomDestination.tsx

```diff
@@ -0,0 +1,147 @@
+import type { PropTypes } from './CustomDestination.types';
+import type { MouseEvent } from 'react';
+
+import { i18n } from '@lingui/core';
+import {
+  Alert,
+  ChevronDownIcon,
+  CloseIcon,
+  Divider,
+  IconButton,
+  PasteIcon,
+  Typography,
+  WalletIcon,
+} from '@rango-dev/ui';
+import React, { useEffect, useRef } from 'react';
+
+import { useAppStore } from '../../store/AppStore';
+import { useQuoteStore } from '../../store/quote';
+import { getBlockchainDisplayNameFor } from '../../utils/meta';
+import { isValidAddress } from '../ConfirmWalletsModal/ConfirmWallets.helpers';
+import { CustomCollapsible } from '../CustomCollapsible/CustomCollapsible';
+import { ExpandedIcon } from '../CustomCollapsible/CustomCollapsible.styles';
+
+import {
+  Container,
+  CustomDestinationButton,
+  StyledTextField,
+} from './CustomDestination.styles';
+
+export function CustomDestination(props: PropTypes) {
+  const { blockchain, handleOpenChange, open } = props;
+
+  const { customDestination, setCustomDestination } = useQuoteStore();
+  const { config } = useAppStore();
+  const blockchains = useAppStore().blockchains();
+  const blockchainName = getBlockchainDisplayNameFor(
+    blockchain.name,
+    blockchains
+  );
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const isFirefox = navigator?.userAgent.includes('Firefox');
+  const isAddressChecked = open && !!customDestination && blockchain;
+  const isAddressInvalid =
+    isAddressChecked && !isValidAddress(blockchain, customDestination);
+  const handleClear = () => {
+    setCustomDestination('');
+  };
+
+  const handlePaste = async (event: MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    if (navigator.clipboard !== undefined) {
+      const pastedText = await navigator.clipboard.readText();
+      setCustomDestination(pastedText);
+      inputRef?.current?.focus();
+    }
+  };
+
+  const renderSuffix = () => {
+    if (customDestination) {
+      return (
+        <IconButton onClick={handleClear} variant="ghost">
+          <CloseIcon size={12} color="gray" />
+        </IconButton>
+      );
+    } else if (!isFirefox) {
+      return (
+        <IconButton onClick={handlePaste} variant="ghost">
+          <PasteIcon size={16} />
+        </IconButton>
+      );
+    }
+
+    return null;
+  };
+
+  useEffect(() => {
+    if (
+      config?.defaultCustomDestinations?.[blockchain.name] &&
+      customDestination === null
+    ) {
+      const destination = config.defaultCustomDestinations[blockchain.name];
+      setCustomDestination(destination);
+      handleOpenChange(true);
+    }
+  }, []);
```

Comment:
You are not passing dependencies to `useEffect`. Are you sure this won't cause any bugs?

Reply:
This indicates the effect only runs once after the mounting of the component, and this is what we require.

Reply:
I suggest considering changing the configuration from outside the widget. Declare the `destination` variable outside the `useEffect` to avoid duplicating code, and include it in the dependency array of `useEffect`.

Reply:
Fixed